app/controllers/idv/confirmations_controller.rb

```diff
@@ -19,8 +19,13 @@ def update
     end
 
     def download
-      data = user_session[:personal_key] + "\r\n"
-      send_data data, filename: 'personal_key.txt'
+      if user_session[:personal_key].present?
+        data = user_session[:personal_key] + "\r\n"
+        send_data data, filename: 'personal_key.txt'
+      else
+        Rails.logger.warn('no personal_key in user_session')
```

Comment:
Should we make an analytics event for the pkey download? That way we'd get all the goodies that come along with that, like session and user IDs.

Reply:
I like that idea!

Reply:
Good point, will update that and then do `success: true/false` instead of separate events

Reply:
updated in 8ba4ebf520e44d471424838fe480db699687ddf0